src/app/progress/SeasonalRank.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { D2ManifestDefinitions } from '../destiny2/d2-definitions';
+import BungieImage from '../dim-ui/BungieImage';
+import {
+  DestinyCharacterProgressionComponent,
+  DestinySeasonDefinition,
+  DestinyProfileResponse
+} from 'bungie-api-ts/destiny2';
+import Countdown from 'app/dim-ui/Countdown';
+
+export default function SeasonalRank({
+  defs,
+  characterProgressions,
+  season,
+  profileInfo
+}: {
+  defs: D2ManifestDefinitions;
+  characterProgressions: DestinyCharacterProgressionComponent;
+  season: DestinySeasonDefinition | undefined;
+  profileInfo: DestinyProfileResponse;
+}) {
+  const formatter = new Intl.NumberFormat(window.navigator.language);
+
+  // Get season details
+  const seasonNameDisplay = season && season.displayProperties.name;
+  const seasonPassProgressionHash = season && season.seasonPassProgressionHash;
+  const seasonEnd = season && season.endDate;
+  const { seasonHashes } = profileInfo.profile.data!;
```

Comment:
Unfortunately this data can be missing - we have to check that it's present. 

Personal preference, but I tend to think there's no need to destructure a single item.